synapse/handlers/room.py

```diff
@@ -1120,7 +1120,9 @@ async def create_event(
                 400, f"'{preset_config}' is not a valid preset", errcode=Codes.BAD_JSON
             )
 
-        creation_content.update({"creator": creator_id})
+        # MSC2175 removes the creator field from the create event.
+        if not room_version.msc2175_implicit_room_creator:
+            creation_content.update({"creator": creator_id})
```

Comment:
Today's spec says:

> Name | Type | Description
> -- | -- | --
> creator | string | Required: The user_id of the room creator. This is set by the homeserver.

which you could argue means that `creator` shouldn't be user controlled?

But I think leave it as-is.

Reply:
Also while we're here, let's just set the value instead of allocating a dict:

```suggestion
        # MSC2175 removes the creator field from the create event.
        if not room_version.msc2175_implicit_room_creator:
            creation_content["creator"] = creator_id
```